File: DSA/RATNA_KHANNA_590013141/DAY_87_QUESTION_3.c
#include <stdio.h>
#include <stdlib.h>
#include <limits.h>

struct Edge {
    int u, v, w;
};

int main() {
    int V, E, src;
    scanf("%d", &V);
    scanf("%d", &E);
    struct Edge edges[E];
    for (int i = 0; i < E; i++) {
        scanf("%d %d %d", &edges[i].u, &edges[i].v, &edges[i].w);
    }
    scanf("%d", &src);

    int dist[V];
    for (int i = 0; i < V; i++) dist[i] = INT_MAX;
    dist[src] = 0;

    for (int i = 1; i <= V - 1; i++) {
        for (int j = 0; j < E; j++) {
            int u = edges[j].u;
            int v = edges[j].v;
            int w = edges[j].w;
            if (dist[u] != INT_MAX && dist[u] + w < dist[v]) {
                dist[v] = dist[u] + w;
            }
        }
    }

    for (int j = 0; j < E; j++) {
        int u = edges[j].u;
        int v = edges[j].v;
        int w = edges[j].w;
        if (dist[u] != INT_MAX && dist[u] + w < dist[v]) {
            printf("Negative cycle detected\n");
            return 0;
        }
    }

    for (int i = 0; i < V; i++) printf("%d ", dist[i]);
    return 0;
}

File: DSA/AnshulJain_590014934/Day_87_Question_3.py
n=int(input())
m=int(input())
edges=[list(map(int,input().split())) for _ in range(m)]
src=int(input())
dist=[float('inf')]*n
dist[src]=0
for _ in range(n-1):
    for u,v,w in edges:
        if dist[u]!=float('inf') and dist[v]>dist[u]+w:
            dist[v]=dist[u]+w
neg_cycle=False
for u,v,w in edges:
    if dist[u]!=float('inf') and dist[v]>dist[u]+w:
        neg_cycle=True
        break
if neg_cycle:
    print("Negative cycle detected")
else:
    print(dist)
